### PR TITLE
Use file extension to avoid duplicates

### PR DESCRIPTION
**Pull request by kevinemoore**
Appending the file extension to node names for file nodes if the base name appears more than once in the packages and would otherwise produce a name conflict.

### CODE REVIEW

File: quilt/tools/build.py [hunk 1]
```diff
@@ -198,19 +199,26 @@ def _generate_contents(dir_path):
             try:
                 safename = _pythonize_name(nodename)
                 if safename in contents:
-                    print("Warning: duplicate name %r in %s." % (safename, dir_path))
-                    continue
+                    if ext:
+                        safename = "{name}_{ext}".format(name=safename, ext=ext)
```
Comment (dimaryaz) [inline quilt/tools/build.py]: Theoretically, _this_ could be a duplicate, too.

File: quilt/tools/main.py [hunk 2]
```diff
@@ -29,10 +29,14 @@ def main():
...
 
+    generate_p = subparsers.add_parser("generate")
+    generate_p.add_argument("directory", help="Source file directory")
+    generate_p.set_defaults(func=command.generate)
+
     build_p = subparsers.add_parser("build")
     build_p.add_argument("package", type=str, help="Owner/Package Name")
     buildpath_group = build_p.add_mutually_exclusive_group(required=True)
-    buildpath_group.add_argument("-d", "--directory", type=str, help="Source file directory")
+    buildpath_group.add_argument("-a", "--auto", type=str, help="Source file directory")
```
Comment (dimaryaz) [inline quilt/tools/main.py]: I guess this isn't a new bug, I wonder if `-a` should just be a boolean - and then `path` could be used as either build file path or directory?

File: quilt/tools/build.py (hunk 1 above)
Comment (dimaryaz) [inline quilt/tools/build.py]: Ignoring this one? :)

Comment (akarve) [inline quilt/tools/build.py]: do we know that they have same parent directory? if so then OS should enforce unique naming. what am i missing?

File: quilt/tools/main.py [hunk 3]
```diff
@@ -36,7 +36,6 @@ def main():
     build_p = subparsers.add_parser("build")
     build_p.add_argument("package", type=str, help="Owner/Package Name")
     buildpath_group = build_p.add_mutually_exclusive_group(required=True)
-    buildpath_group.add_argument("-a", "--auto", type=str, help="Source file directory")
     buildpath_group.add_argument("path", type=str, nargs='?', help="Path to the Yaml build file")
```
Comment (akarve) [inline quilt/tools/main.py]: help needs a bit of love here in light of new automatic behavior

**Pull request merged**